Make getblockheader works both with block height and block hash

sickpig opened the pull request:
This is basically the same as we did for getblockstats.
Probably we could isolate the code that deal with understanding
if the param is an integer or a string representing a block hash.

sickpig committed:
Make getblockheader works both with block height and block hash

This is basically the same as we did for getblockstats.
Probably we could isolate the code that deal with understanding
if the param is an integer or a string representing a block hash.

ptschip commented:
Looks pretty straightforward...Maybe you could add a couple of simple tests to one of the py scripts, like blockchain.py?

sickpig commented:
> Looks pretty straightforward...Maybe you could add a couple of simple tests to one of the py scripts, like blockchain.py?

good idea, will do.

sickpig commented:
ad WIP tag until I'll manage to add tests

sickpig committed:
Add a further test for getblockheader in blockchain.py

Verify that getblockheader give the same output despite the fact we reference the block by height or by hash

sickpig commented:
test added, taking out of WIP
